PR: Add explanation on the use of subsections in the Outline Explorer (#293)
spyder-ide/spyder-docs: doc/panes/editor.rst

* ``# <codecell>`` (IPython notebook cell separator)

Providing a description to the right of the separator will give that cell its own name in the :guilabel:`Outline Explorer`.

> You can also create "subsections" by adding ``%`` signs to the cell separator, e.g. ``# %%%`` to create a level 2 subsection, ``# %%%%`` for level 3, etc...

Reviewer: ```suggestion
You can also create "subsections" by adding more ``%`` signs to the cell separator, e.g. ``# %%%`` to create a level 2 subsection, ``# %%%%`` for level 3, etc.
```

* Add clarifying "more"
* Tweak ending punctuation so sentence ends in a period.

> This creates multiple outline levels in the the :guilabel:`Outline Explorer`:

Reviewer: ```suggestion
This displays multiple levels in the the :guilabel:`Outline Explorer`.
```

* Avoid word repetition ("create", "outline")
* Swap colon for period (per our convention for introducing figures)

> .. image:: /images/editor/editor-subsections.png

Reviewer: Great job on the screenshot!

   :alt: Spyder outline panel, showing an example of sub sections

> Note that this only has an impact on how the code outline is displayed, it has no effect on the functioning of the code cells.

Reviewer: ```suggestion
Note that this only affects how the outline is displayed; it has no impact on the functioning of the code cells.
```

* Use semicolon instead of comma to avoid a comma splice
* Swap "impact" and "effect" as it simplifies the sentence a bit and read a little cleaner, at least to my ear
* Avoid word repetition of "code"
